haskell/package.yaml

```diff
@@ -21,6 +21,10 @@ verbatim:
 library:
   <<: *lib-common
 
+  ghc-options:
+    - -Wall
```

Comment:
`-Wall` here is useless because all warnings enabled by `-Wall` are already enabled, see `hpack/definitions.yaml`. Having `-Werror` in `.cabal` file is a bad practice because if someone else uses our library as a dependency and tries to build it with a different compiler version, they may be unable to do it because there can be new warnings in the new compiler version. What we actually want to do is to treat warnings as errors on CI, that should be done as part of CI configuration. If you need an example see https://github.com/serokell/tezos-globacap/blob/master/ci.nix